1-callbacks/solution.js

```diff
@@ -34,14 +34,46 @@ node solution.js name1 name2 name3
 
 function solution() {
     // YOUR SOLUTION GOES HERE
+    const validateUser = require('./validate-user');
 
     // you get your 5 names here
-
-    // iterate the names array and validate them with the method
-
-    // log the final result
+    const names = 
+        process.argv[2] ?
+            Array.from(process.argv.slice(2))
+            : ['Christine', 'Carrie', 'Richard', 'Carol', 'Stacy'];
+
+    //iterate the names array and validate them with the method
+    //log the final result
+    console.log('Failure\n');
+    names.map(name => {
+        validateUser(name, (error, user) => {
+            error ? 
+                console.log(error.message)
+                : console.log(`\nid:${user.id}\nname: ${user.name}`);
```

Comment:
validate existence of error using if condition

Reply:
Which considerations should we take using map?

Reply:
you're  right, my mistake in using map instead other loop like forEach because map, in this case, is reducing performance.